File: PGS/기능개발.py
def solution(progresses, speeds):
    answer = []
    day = []
    
    for i in range(len(progresses)):
        t = 100-progresses[i]
        if t%speeds[i]==0:
            day.append(t//speeds[i])
        else:
            day.append(t//speeds[i] + 1)
            
    
    d = day[0]
    cnt = 1
    for i in range(1, len(day)):
        if d >= day[i]:
            cnt += 1
        else:
            answer.append(cnt)
            cnt = 1
            d = day[i]
    answer.append(cnt)
    
    return answer

File: PGS/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> answer = new ArrayList<>();
        int[] day = new int[progresses.length];
        Deque<Integer> dq = new ArrayDeque<>();
        
        for(int i=0; i<progresses.length; i++){
            
            if((100-progresses[i])%speeds[i]==0){
                day[i] = (100-progresses[i])/speeds[i];
            }else{
                day[i] = (100-progresses[i])/speeds[i]+1;
            }
        }
        
        int first=0;
        for(int i=0; i<day.length; i++){
            if(dq.isEmpty()){
                dq.addLast(day[i]);
            }else{ //있으면 
                if(dq.peekFirst() < day[i]){
                    answer.add(dq.size());
                    dq.clear();
                }
                dq.addLast(day[i]);
            }
        }
        if(!dq.isEmpty()){
            answer.add(dq.size());
        }
        
        return answer.stream().mapToInt(i->i).toArray();
    }
}